### PR TITLE
Add configurable product selectors and facade

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/product/src/selectors/configurable-product/configurable-product.selectors.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	return flag;
+}
+
+function getMinimumPrice(variants: DaffConfigurableProductVariant[]): string {
+	let price: number = parseInt(variants[0].price, 10);
+	variants.forEach(variant => {
+		if(parseInt(variant.price, 10) < price) price = parseInt(variant.price, 10);
+	});
+
+	return price.toString();
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/src/selectors/configurable-product/configurable-product.selectors.ts]: done

File: libs/product/src/selectors/configurable-product/configurable-product.selectors.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	appliedAttributes: DaffProductVariantAttributesDictionary, 
+	variant: DaffConfigurableProductVariant
+): boolean {
+	let flag = true;
+	configurableAttributes.forEach(option => {
+		if(appliedAttributes[option.code] && variant.appliedAttributes[option.code] !== appliedAttributes[option.code]) {
+			flag = false;
+		}
+	});
+	return flag;
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/src/selectors/configurable-product/configurable-product.selectors.ts]: done. I don't think the boolean logic can be simplified, but I added some private function with descriptive names to try to make it more readable.

File: libs/product/src/selectors/configurable-product/configurable-product.selectors.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	return price.toString();
+}
+
+function getMaximumPrice(variants: DaffConfigurableProductVariant[]): string {
+	let price: number = parseInt(variants[0].price, 10);
+	variants.forEach(variant => {
+		if(parseInt(variant.price, 10) > price) price = parseInt(variant.price, 10);
+	});
+
+	return price.toString();
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply libs/product/src/selectors/configurable-product/configurable-product.selectors.ts]: done

File: libs/product/testing/src/helpers/mock-configurable-product-facade.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { BehaviorSubject } from 'rxjs';
+import { Dictionary } from '@ngrx/entity';
+
+import { DaffConfigurableProductFacadeInterface } from '@daffodil/product';
+
+export class MockDaffConfigurableProductFacade implements DaffConfigurableProductFacadeInterface {
+	getAppliedAttributes(id: string): BehaviorSubject<Dictionary<string>> {
+		return new BehaviorSubject(null);
```
Comment (griest024) [inline libs/product/testing/src/helpers/mock-configurable-product-facade.ts:8]: I think this should be a `BehaviorSubject({})` to better match the return type.

File: libs/product/testing/src/helpers/mock-configurable-product-facade.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	};
+	getPrice(id: string): BehaviorSubject<string> {
+		return new BehaviorSubject(null);
+	};
+	getUndeterminedAttributes(id: string): BehaviorSubject<Dictionary<string[]>> {
+		return new BehaviorSubject(null);
```
Comment (griest024) [inline libs/product/testing/src/helpers/mock-configurable-product-facade.ts:14]: I think this should be a `BehaviorSubject({})` to better match the return type.

File: libs/product/testing/src/helpers/mock-configurable-product-facade.ts (hunk 4 above)
Comment (lderrickable) [reply libs/product/testing/src/helpers/mock-configurable-product-facade.ts]: changed

File: libs/product/testing/src/helpers/mock-configurable-product-facade.ts (hunk 5 above)
Comment (lderrickable) [reply libs/product/testing/src/helpers/mock-configurable-product-facade.ts]: changed

**Pull request merged**